### PR TITLE
Use `showBlock` in applicant service 

### PR DESCRIPTION
**Pull request by carolinedanzi**
### Description
1. `getInProgressBlocks()` now returns blocks that should be shown to the user based on predicates
2. `getAllBlocks()` returns all blocks in a program, except those that were hidden from the applicant. This allows us to avoid showing or submitting blocks in the applicant summary or application submission that the applicant didn't complete or did not need to complete

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
#1001

### CODE REVIEW

**Review by hmhsu — APPROVED** [review]

**Comment by carolinedanzi** [comment]
Note: converting to draft - ran locally and found the block is not shown properly and appears in the application review. This prevents submitting, since the application is not complete

File: universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java [hunk 1]
```diff
@@ -24,7 +24,13 @@
 
 public class ReadOnlyApplicantProgramServiceImpl implements ReadOnlyApplicantProgramService {
 
+  /**
+   * Note that even though {@link ApplicantData} is mutable, we can consider it immutable at this
```
[the remark this answers is not in the document]
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java:28]: Great point! Linked to the lock method and updated comment. Thanks!

File: universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramService.java [hunk 2]
```diff
@@ -10,8 +10,8 @@
 
   /**
    * Get the {@link Block}s for this program and applicant. This includes all blocks, whether the
-   * block was filled out in this program or a previous program. This also includes blocks that may
-   * be hidden from the applicant (i.e. they have a show/hide predicate).
+   * block was filled out in this program or a previous program. This will not include blocks that
+   * are hidden from the applicant (i.e. they have a show/hide predicate).
    */
   ImmutableList<Block> getAllBlocks();
```
Comment (hmhsu) [inline universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramService.java:16]: Suggestion (non-blocking): I feel that `getAllBlocks` not technically returning all blocks a bit confusing here. Can we name this something like `getAllActiveBlocks`?

**Review by hmhsu — APPROVED** [review]
Looks good with a suggestion in naming.

File: universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramService.java (hunk 2 above)
Comment (carolinedanzi) [reply universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramService.java:16]: I think you're right, I like `getAllActiveBlocks` a lot more. Will update!

**Pull request merged**